ecommerce/api.py

```diff
@@ -107,6 +109,37 @@ def generate_checkout_payload(request):
     return payload
 
 
+def check_discount_for_products(discount, basket):
+    """
+    Checks the validity of the discount against what's in the basket.
+
+    If the discount either has no products associated with it, or the products
+    in the basket are applicable to the discount, this returns True.
+    Otherwise, returns False.
+
+    Args:
+        - basket (Basket): the current basket
+        - discount (Discount|string: the discount to apply (if a string, loads the discount code specified)
+    Returns:
+        boolean
+    """
+    if not isinstance(discount, Discount):
+        discount = Discount.objects.filter(discount_code=discount).first()
+
+    basket_item_products = basket.get_products()
+
+    if (
+        not discount.products.count() == 0
+        and DiscountProduct.objects.filter(product__in=basket_item_products)
+        .filter(discount=discount)
+        .count()
```

Comment:
I think calling .exits() would be a little cleaner here

Reply:
Also you could rewrite this if statement and returns as:
```
return discount.products.count() == 0 or not DiscountProduct.objects.filter(product__in=basket_item_products)
        .filter(discount=discount).exists()
```

Reply:
Done in https://github.com/mitodl/mitxonline/pull/708/commits/fe65eb05b4c6f846d4b157a46804f0f800656674